### PR TITLE
Pulsar binder basic integ test

### PR DESCRIPTION
**Pull request by sobychacko**

### CODE REVIEW

File: spring-pulsar-spring-cloud-stream-binder/build.gradle [hunk 1]
```diff
@@ -11,5 +11,9 @@ dependencies {
         exclude group: 'javax.annotation', module: 'javax.annotation-api'
     }
     testImplementation 'org.springframework.boot:spring-boot-starter-test'
+    testImplementation project(':spring-pulsar-test')
+    testImplementation 'org.testcontainers:pulsar'
+    testImplementation 'org.awaitility:awaitility'
```
Comment (onobc) [inline spring-pulsar-spring-cloud-stream-binder/build.gradle:16]: **[NIT]** Do you mind alpha ordering the last 3 of these newly added deps?

File: spring-pulsar-spring-cloud-stream-binder/src/test/java/org/springframework/pulsar/spring/cloud/stream/binder/PulsarBinderIntegrationTests.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import org.springframework.boot.SpringBootConfiguration;
+import org.springframework.boot.WebApplicationType;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.system.CapturedOutput;
+import org.springframework.boot.test.system.OutputCaptureExtension;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.pulsar.test.support.PulsarTestContainerSupport;
+
+@ExtendWith(OutputCaptureExtension.class)
```
Comment (onobc) [inline spring-pulsar-spring-cloud-stream-binder/src/test/java/org/springframework/pulsar/spring/cloud/stream/binder/PulsarBinderIntegrationTests.java:39]: `@author` :)

File: spring-pulsar-spring-cloud-stream-binder/src/test/java/org/springframework/pulsar/spring/cloud/stream/binder/PulsarBinderIntegrationTests.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+	@Test
+	void basicProducerConsumerBindingEndToEnd(CapturedOutput output) {
+		SpringApplication app = new SpringApplication(BasicScenarioConfig.class);
+		app.setWebApplicationType(WebApplicationType.NONE);
+		try (ConfigurableApplicationContext context = app.run(
+				"--spring.pulsar.client.service-url=" + PulsarTestContainerSupport.getPulsarBrokerUrl(),
+				"--spring.cloud.function.definition=textSupplier;textLogger",
+				"--spring.cloud.stream.bindings.textLogger-in-0.destination=textSupplier-out-0",
+				"--spring.cloud.stream.pulsar.bindings.textLogger-in-0.consumer.subscription-name=basic-scenario-sub-1")) {
```
Comment (onobc) [inline spring-pulsar-spring-cloud-stream-binder/src/test/java/org/springframework/pulsar/spring/cloud/stream/binder/PulsarBinderIntegrationTests.java:50]: Can we auto-generate the sub name if user does not specify one?

File: spring-pulsar-spring-cloud-stream-binder/build.gradle [hunk 4]
```diff
@@ -11,5 +11,9 @@ dependencies {
         exclude group: 'javax.annotation', module: 'javax.annotation-api'
     }
     testImplementation 'org.springframework.boot:spring-boot-starter-test'
+    testImplementation project(':spring-pulsar-test')
+    testImplementation 'org.testcontainers:pulsar'
```
Comment (onobc) [inline spring-pulsar-spring-cloud-stream-binder/build.gradle:15]: If we change these 2 deps to `api` in the `spring-test` lib then all users of the lib can remove these 2 deps. 

Don't worry about that in this PR though - just noticed it.

**Review by onobc — CHANGES_REQUESTED** [review]
@sobychacko nice test! A few minor comments to address, but other than that LGTM.

File: spring-pulsar-spring-cloud-stream-binder/src/test/java/org/springframework/pulsar/spring/cloud/stream/binder/PulsarBinderIntegrationTests.java (hunk 3 above)
Comment (sobychacko) [reply spring-pulsar-spring-cloud-stream-binder/src/test/java/org/springframework/pulsar/spring/cloud/stream/binder/PulsarBinderIntegrationTests.java:50]: Yes, let's do that later outside of this PR. I was already thinking about that.

**Pull request merged**